### PR TITLE
Change default behavior of gateway login module

### PR DESCRIPTION
**Pull request by justinma246**
Login module passes when there is a realm and login module. No require-role / require-valid user. Ticket [#825 https://github.com/kaazing/roadmap/issues/825]

### CODE REVIEW

File: server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java [hunk 1]
```diff
@@ -512,11 +512,23 @@ private DefaultServiceDefaultsContext resolveServiceDefaults(ServiceDefaultsType
...
             }
-
+            
+            
+            
             Collection<String> requireRolesCollection = new LinkedList<>();
             for (AuthorizationConstraintType authConstraint : serviceConfig.getAuthorizationConstraintArray()) {
                 Collections.addAll(requireRolesCollection, authConstraint.getRequireRoleArray());
             }
+            RealmContext realmContext = null;
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java]: Good!

File: server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java [hunk 2]
```diff
@@ -512,11 +512,23 @@ private DefaultServiceDefaultsContext resolveServiceDefaults(ServiceDefaultsType
                 properties.remove("connect");
                 connectURIs.add(resolveURI(getCanonicalURI(connectProperty, true)));
             }
-
+            
+
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java]: nit: please remove whitespace

File: transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestIT.java [hunk 3]
```diff
@@ -37,9 +37,9 @@
                     .accept("ws://localhost:8001/echoAuth")
                     .type("echo")
                     .realmName("demo")
-                        .authorization()
-                        .requireRole("USER")
-                    .done()
+//                        .authorization()
```
Comment (dpwspoon) [inline transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestIT.java]: We should remove these comments, or duplicate the test and have one with the comments removed and one with them there

**Comment by dpwspoon** [comment]
@justinma246 is this a duplicate of https://github.com/kaazing/gateway/pull/604

**Comment by justinma246** [comment]
Yes, sorry, I thought i had closed this

File: server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java [hunk 4]
```diff
@@ -511,12 +511,22 @@ private DefaultServiceDefaultsContext resolveServiceDefaults(ServiceDefaultsType
...
             Collection<String> requireRolesCollection = new LinkedList<>();
             for (AuthorizationConstraintType authConstraint : serviceConfig.getAuthorizationConstraintArray()) {
                 Collections.addAll(requireRolesCollection, authConstraint.getRequireRoleArray());
             }
+            RealmContext realmContext = null;
+            String name = serviceConfig.getRealmName();
+            if (serviceConfig.isSetRealmName()) {
+                realmContext = realmsContext.getRealmContext(name);
+                if (realmContext != null) {
+                    if (requireRolesCollection.size() == 0) {
```
Comment (jitsni) [inline server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java]: You can use requireRolesCollection.isEmpty()

File: transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestITRole.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class BasicLoginModuleTestITRole {
```
Comment (jitsni) [inline transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestITRole.java]: You want to rename the file. It should end with IT.java
May be BasicLoginModuleRoleIT.java

File: transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestITRole.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                .done()
+                .security()
+                    .realm()
+                        .name("demo")
+                        .description("Kaazing WebSocket Gateway Demo")
+                        .httpChallengeScheme("Basic")
+                        .loginModule()
+                            .type("class:org.kaazing.gateway.transport.wsn.auth.TestBasicLoginModule")
+                            .success("requisite")
+                          .option("roles", "USER")
```
Comment (jitsni) [inline transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/BasicLoginModuleTestITRole.java]: formatting for .option line ?

File: server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java [hunk 7]
```diff
@@ -511,12 +511,22 @@ private DefaultServiceDefaultsContext resolveServiceDefaults(ServiceDefaultsType
                 connectProperty = connectProperty.trim();
                 properties.remove("connect");
                 connectURIs.add(resolveURI(getCanonicalURI(connectProperty, true)));
-            }
-
+            }         
+
```
Comment (jitsni) [inline server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java]: unnecessary space is added.

**Comment by justinma246** [comment]
Edited accordingly to @jitsni comments

**Comment by justinma246** [comment]
This was outdated - I made the changes in ticket #604

**Pull request closed without merging**